camel/toolkits/linkedin_toolkit.py

```diff
@@ -0,0 +1,258 @@
+        }
+
+        response = requests.get(
+            "https://api.linkedin.com/v2/userinfo",
+            headers=headers,
+        )
+
+        if response.status_code != HTTPStatus.OK:
+            raise Exception(
+                f"Failed to retrieve profile. "
+                f"Status code: {response.status_code}, "
+                f"Response: {response.text}"
+            )
+
+        json_response = response.json()
+
+        locale = json_response.get('locale', {})
+        country = locale.get('country', 'N/A')
+        language = locale.get('language', 'N/A')
+
+        profile_report = {
+            "Country": country,
+            "Language": language,
+            "First Name": json_response.get('given_name'),
+            "Last Name": json_response.get('family_name'),
+            "Email": json_response.get('email'),
+        }
+
+        if include_id:
+            profile_report['id'] = f"urn:li:person:{json_response['sub']}"
+
+        return profile_report
+
+    def get_tools(self) -> List[OpenAIFunction]:
+        r"""Returns a list of OpenAIFunction objects representing the
+        functions in the toolkit.
+
+        Returns:
+            List[OpenAIFunction]: A list of OpenAIFunction objects
+                representing the functions in the toolkit.
+        """
+        return [
+            OpenAIFunction(self.create_post),
+            OpenAIFunction(self.delete_post),
+            OpenAIFunction(self.get_my_profile),
+        ]
+
+    def _get_access_token(self) -> str:
+        r"""Fetches the access token required for making LinkedIn API requests.
+
+        Returns:
+            str: The OAuth 2.0 access token.
+
+        Raises:
+            EnvironmentError: If the environment variable
+                              `LINKEDIN_ACCESS_TOKEN` is not set or is empty.
+        """
+        token = os.getenv("LINKEDIN_ACCESS_TOKEN", "")
+        if not token:
+            return "Access token not found. Please set LINKEDIN_ACCESS_TOKEN."
+        return token
+
+    def _handle_http_error(self, response: requests.Response) -> str:
+        r"""Handles the HTTP errors based on the status code of the response.
+
+        Args:
+            response (requests.Response): The HTTP response from the API call.
+
+        Returns:
+            str: The error type, based on the status code.
+        """
+        if response.status_code == HTTPStatus.UNAUTHORIZED:
+            return "Unauthorized. Check your access token."
+        elif response.status_code == HTTPStatus.FORBIDDEN:
+            return (
+                "Forbidden. You do not have permission to perform this action."
+            )
+        elif response.status_code == HTTPStatus.NOT_FOUND:
+            return "Not Found. The resource could not be located."
+        elif response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+            return "Too Many Requests. You have hit the rate limit."
+        else:
+            return "HTTP Error"
```

Comment:
@Wendong-Fan hi, This function seems to be very versatile, can it be placed in utils? What do you think? Thank you.